pkg/storaged/btrfs/btrfs-tool.py

```diff
@@ -95,59 +63,47 @@ def list_filesystems():
 tmp_mountpoints = set()
 
 
-def add_tmp_mountpoint(db, fs, dev, opt_repair):
+def get_tmp_mountpoint_path(uuid):
+    return os.path.join(TMP_MP_DIR, str(os.getpid()) + ":" + uuid)
+
+
+def add_tmp_mountpoint(fs):
     global tmp_mountpoints
     uuid = fs['uuid']
-    if uuid not in tmp_mountpoints:
-        debug(f"ADDING {uuid}")
-        tmp_mountpoints.add(uuid)
-        if uuid in db and db[uuid] > 0:
-            db[uuid] += 1
-        else:
-            db[uuid] = 1
-        if not fs['has_tmp_mountpoint'] and (db[uuid] == 1 or opt_repair):
-            path = os.path.join(TMP_MP_DIR, uuid)
-            debug(f"MOUNTING {path}")
-            os.makedirs(path, exist_ok=True)
-            subprocess.check_call(["mount", dev, path])
+    tmp_mountpoints.add(uuid)
+    path = get_tmp_mountpoint_path(uuid)
+    debug(f"MOUNTING {path}")
+    os.makedirs(path, exist_ok=True)
+    subprocess.check_call(["mount", fs['devices'][0], path])
+    return path
 
 
-def remove_tmp_mountpoint(db, uuid):
+def remove_tmp_mountpoint(uuid):
     global tmp_mountpoints
     if uuid in tmp_mountpoints:
-        debug(f"REMOVING {uuid}")
         tmp_mountpoints.remove(uuid)
-        if db[uuid] == 1:
-            path = os.path.join(TMP_MP_DIR, uuid)
-            try:
-                debug(f"UNMOUNTING {path}")
-                subprocess.check_call(["umount", path])
-                subprocess.check_call(["rmdir", path])
-            except Exception as err:
-                sys.stderr.write(f"Failed to unmount {path}: {err}\n")
-            del db[uuid]
-        else:
-            db[uuid] -= 1
+        path = get_tmp_mountpoint_path(uuid)
+        try:
+            debug(f"UNMOUNTING {path}")
+            subprocess.check_call(["umount", path])
+            subprocess.check_call(["rmdir", path])
+        except Exception as err:
+            sys.stderr.write(f"Failed to unmount {path}: {err}\n")
 
 
 def remove_all_tmp_mountpoints():
-    if len(tmp_mountpoints) > 0:
-        with mount_database() as db:
-            for mp in set(tmp_mountpoints):
-                remove_tmp_mountpoint(db, mp)
-
+    for mp in set(tmp_mountpoints):
+        remove_tmp_mountpoint(mp)
```

Comment:
Even this could be dropped. There will only be a handful of directories, they cost next to nothing, and now that they are in /run they won't survive a reboot anyway. (Fine to keep it, just thinking aloud)

Reply:
Yes, I was thinking the same, just wanted to produce a version that does a full cleanup first.